scripts/us_census/pep/pep_by_asrh/preprocess.py

```diff
@@ -0,0 +1,1072 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This Python Script Load the datasets, cleans it
+and generates cleaned csv, MCF, TMCF file.
+"""
+import os
+
+from copy import deepcopy
+import pandas as pd
+
+from absl import app
+from absl import flags
+from cols_map import _get_mapper_cols_dict
+
+FLAGS = flags.FLAGS
+default_input_path = os.path.dirname(
+    os.path.abspath(__file__)) + os.sep + "ip_data"
+flags.DEFINE_string("input_path", default_input_path, "Import Data File's List")
+
+
+def _convert_to_int(data_df: pd.DataFrame) -> pd.DataFrame:
+    for col in data_df.columns:
+        data_df[col] = data_df[col].astype("int", errors="ignore")
+    return data_df
+
+
+def _derive_cols(data_df: pd.DataFrame, derived_cols: dict) -> pd.DataFrame:
+    """Derive new columns using DataFrame and derived_cols dict
+    Args:
+        data_df (pd.DataFrame): Input DataFrame loaded with data.
+        derived_cols (dict): derived_cols dict
+       where key represents dervied col and values represents existing columns.
+
+    Returns:
+        pd.DataFrame: DataFrame
+    """
+    for dsv, sv in derived_cols.items():
+        data_df[dsv] = data_df.loc[:, sv].apply(pd.to_numeric,
```

Comment:
Is it possible for this to run in to an error? You are also using "coerce" which is risky. Could you document what exactly is going on here? Why should we not be catching the errors here?

Reply:
Here, we are converting numerical string values to int datatype. Removed coerce option in the conversion